ibus-engines.mozc: 2.26.4660.102 -> 2.29.5268.102

pineapplehunter opened the pull request:
## Description of changes
Update the ibus mozc package to version 2.29.5268.102.
This changes the build system from mkDerivation to buildBazelPackage.

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

kirillrdy commented:
@pineapplehunter please fix up commit message and PR title

pineapplehunter commented:
@kirillrdy updated the commit message and PR title.

kirillrdy commented on pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix:
```diff
@@ -1,89 +1,62 @@
     repo = "mozc";
-    rev = "refs/tags/${version}";
-    hash = "sha256-sgsfJZALpPHFB5bXu4OkRssViRDaPcgLfEyGhbqvJbI=";
+    rev = "${version}";
```
```suggestion
    rev = version;
```

kirillrdy commented on pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix:
```diff
@@ -1,89 +1,62 @@
-    install    -m 644 data/images/unix/ui-alpha_full.png    $out/share/ibus-mozc/alpha_full.png
-    install -D -m 755 out_linux/Release/mozc_renderer       $out/lib/mozc/mozc_renderer
-    install -D -m 755 out_linux/Release/mozc_emacs_helper   $out/lib/mozc/mozc_emacs_helper
+  preConfigure = ''
```
sourceRoot = "${src.name}/src";

pineapplehunter replied:
In a previous PR I was told that using cd is better.
https://github.com/NixOS/nixpkgs/pull/271379#discussion_r1421921544

kirillrdy replied:
good to know, someone should update nixpkgs manual

I can understand why people don't want to touch `sourceRoot`

kirillrdy commented on pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix:
```diff
@@ -1,89 +1,62 @@
-
-    runHook postBuild
+  postPatch = ''
+    substituteInPlace src/config.bzl \
```
```suggestion
    substituteInPlace config.bzl \
```

pineapplehunter replied:
I will merge this into preConfigure

kirillrdy replied:
given your comment about `cd` keep this as is

kirillrdy reviewed (commented):
![image](https://github.com/NixOS/nixpkgs/assets/12160/daab47ce-08ad-43b8-aaf1-f84e9a761dee)

it works !

kirillrdy commented on pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix:
```diff
@@ -1,89 +1,62 @@
 
-    runHook postInstall
+  buildAttrs.installPhase = ''
+    unzip bazel-bin/unix/mozc.zip -x "tmp/*" -d /
```
we should run preInstall and postInstall hooks

pineapplehunter replied:
updated the commit

kirillrdy commented:
build fails on aarch64-linux

lets wait to see what ofborg says

```
error: builder for '/nix/store/bdprlbn73s77gdgs7vj67yk8jsf6p0dm-ibus-mozc-2.29.5268.102.drv' failed with exit code 1;
       last 10 log lines:
       > Analyzing: target //:package (53 packages loaded, 302 targets configured)
       > Analyzing: target //:package (60 packages loaded, 302 targets configured)
       > Analyzing: target //:package (60 packages loaded, 302 targets configured)
       > Analyzing: target //:package (62 packages loaded, 330 targets configured)
       > ERROR: /build/source/src/renderer/qt/BUILD.bazel:77:12: no such target '@qt_linux//:libexec/moc': target 'libexec/moc' not declared in package '' defined by /build/output/external/qt_linux/BUILD.bazel (Tip: use `query "@qt_linux//:*"` to see all the targets in that package) and referenced by '//renderer/qt:moc_qt_server'
       > ERROR: Analysis of target '//:package' failed; build aborted:
       > INFO: Elapsed time: 50.344s
       > INFO: 0 processes.
       > FAILED: Build did NOT complete successfully (76 packages loaded, 338 targets configured)
       > 
       For full logs, run 'nix log /nix/store/bdprlbn73s77gdgs7vj67yk8jsf6p0dm-ibus-mozc-2.29.5268.102.drv'.
error: 1 dependencies of derivation '/nix/store/pwb7wpimd1nm1kw033ivdsqv9aqap4yp-emacs-mozc-20231026.1200.drv' failed to build
error: 1 dependencies of derivation '/nix/store/vkq89abp9xz5h97q3vc5sa50laxy7800-emacs-ac-mozc-20150227.1619.drv' failed to build
error: 1 dependencies of derivation '/nix/store/pqcxvvg2d7mc00aq5005jc65ihw3rgps-emacs-mozc-cand-posframe-20200208.750.drv' failed to build
error: 1 dependencies of derivation '/nix/store/cqyjrr3nm21hbbi29fawfl9ggl8kyis2-emacs-mozc-im-20160412.22.drv' failed to build
error: 1 dependencies of derivation '/nix/store/fk5g6amj840ypn6hy2abr6p7imkc9yqp-emacs-mozc-popup-20150224.34.drv' failed to build
error: 1 dependencies of derivation '/nix/store/k80nhknrvp3m02g3q56bfjkjwpymz35f-emacs-mozc-temp-20160228.840.drv' failed to build
error: 7 dependencies of derivation '/nix/store/2plqh6680r15fcd1pfbspx2284f3vavm-review-shell.drv' failed to build

```

kirillrdy commented:
@pineapplehunter I think we can change platforms to x86_64-linux, unless there is an easy fix for aarch64-linux

also can you split version change from adding yourself as maintainer into separate commits ?

pineapplehunter committed:
ibus-engines.mozc: add maintainer pineapplehunter

pineapplehunter committed:
ibus-engines.mozc: 2.26.4660.102 -> 2.29.5268.102

pineapplehunter commented:
@kirillrdy I splitted the commits and changed the platform.
It's too bad this doesn't work on aarch64 platforms, since there seems to be more work on them. I guess it is a matter of having the right hash for the platform, so it could be fixed by,
- having multiple hashes for each platform
- changing buildBazelPackage fetcher so it downloades packages for all platforms

Unfortunately I dont have any aarch64-linux machines around to test, so if someone comes along and fixes it that would be nice.

kirillrdy commented:
@pineapplehunter I don't mind loosing aarch64-linux support but I am a bit concerned about buildBazelPackage, its not very widely used, it's not event mentioned in nixpkgs manual.

maybe @NickCao @wegank can comment on its usage

@pineapplehunter as for access to aarch64-linux, if you run nixos you can use qemu to allow running aarch64-linux
`
boot.binfmt.emulatedSystems = [ "aarch64-linux" ]
`

then `nix-build -A ibus-engines.mozc --argstr system aarch64-linux`

kirillrdy commented:
@pineapplehunter thank you for working on this, its better to merge this than keeping it broken

kirillrdy commented:
hmm doesnt build anymore :-) 

```
error: hash mismatch in fixed-output derivation '/nix/store/a6z9zm8cbp3wg7fi8b58npjgls3g6f74-ibus-mozc-2.29.5268.102-deps.tar.gz.drv':
         specified: sha256-S0xB+XujZM2dFMHzVytfgbX9VoMqiXsmPCJaCVR9MH4=
            got:    sha256-nySbIIeic1znPFm0xL2n4TNBIjDMSLgXHucC2p5BdtE=
```

pineapplehunter commented:
@kirillrdy Hmm. That's a problem...
Is it better to have the older version that uses the other build method and keep working on this until we can be sure that it is reproducable?

pineapplehunter commented:
I updated the PR with a new hash so it should be compiling now.

kirillrdy commented:
> @kirillrdy Hmm. That's a problem...
> Is it better to have the older version that uses the other build method and keep working on this until we can be sure that it is reproducable?

I've tried  `nix-build -A ibus-engines.mozc.deps --check` several times and hash is stable for now, it would be good if we had a copy of old deps derivation so that we could see what changed

> Is it better to have the older version that uses the other build method and keep working on this until we can be sure that it is reproducable?

I'll leave it up-to you to decide, we will have to update to newer version sooner or later and gyp builds are no longer supported

pineapplehunter commented:
@kirillrdy 
> I've tried nix-build -A ibus-engines.mozc.deps --check several times and hash is stable for now, it would be good if we had a copy of old deps derivation so that we could see what changed

I found an older copy I had around and checked the difference. It seems like the dependencies have the hash of some packages, wich means that the hash will change if there's any update to the dep inputs.

<details>
  <summary>Diff from an older version</summary>

```diff
diff --git a/a/external/ibus/BUILD.bazel b/b/external/ibus/BUILD.bazel
index 17b5104..5b22f98 100755
--- a/a/external/ibus/BUILD.bazel
+++ b/b/external/ibus/BUILD.bazel
@@ -6,19 +6,19 @@ package(
 cc_library(
     name = "ibus",
     hdrs = glob([
-        "nix/store/hvpil7yyqfiv87rgzvzjrq41daa182my-ibus-1.5.28-dev/include/ibus-1.0/**",
-        "nix/store/nynq8wryq2lzq92mcj0niiwady33qhss-glib-2.78.3/lib/glib-2.0/include/**",
-        "nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include/**",
-        "nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include/glib-2.0/**"
+        "nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include/**",
+        "nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include/glib-2.0/**",
+        "nix/store/1s7z43yp594lrz7ymhvyj1b9g8b8mg0y-ibus-1.5.28-dev/include/ibus-1.0/**",
+        "nix/store/kl4j6fjadg3lqaqv8r9p1lsjrwmi4cz8-glib-2.78.3/lib/glib-2.0/include/**"
     ]),
     copts = [
         ""
     ],
     includes = [
-        "nix/store/hvpil7yyqfiv87rgzvzjrq41daa182my-ibus-1.5.28-dev/include/ibus-1.0",
-        "nix/store/nynq8wryq2lzq92mcj0niiwady33qhss-glib-2.78.3/lib/glib-2.0/include",
-        "nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include",
-        "nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include/glib-2.0"
+        "nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include",
+        "nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include/glib-2.0",
+        "nix/store/1s7z43yp594lrz7ymhvyj1b9g8b8mg0y-ibus-1.5.28-dev/include/ibus-1.0",
+        "nix/store/kl4j6fjadg3lqaqv8r9p1lsjrwmi4cz8-glib-2.78.3/lib/glib-2.0/include"
     ],
     linkopts = [
         "-lgio-2.0",
diff --git a/b/external/ibus/nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include b/b/external/ibus/nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include
new file mode 120000
index 0000000..980767b
--- /dev/null
+++ b/b/external/ibus/nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include
@@ -0,0 +1 @@
+/nix/store/06bn3nr5b1fhz70sdrvr2ccyndpdriib-glib-2.78.3-dev/include
\ No newline at end of file
diff --git a/b/external/ibus/nix/store/1s7z43yp594lrz7ymhvyj1b9g8b8mg0y-ibus-1.5.28-dev/include/ibus-1.0 b/b/external/ibus/nix/store/1s7z43yp594lrz7ymhvyj1b9g8b8mg0y-ibus-1.5.28-dev/include/ibus-1.0
new file mode 120000
index 0000000..add8bc8
--- /dev/null
+++ b/b/external/ibus/nix/store/1s7z43yp594lrz7ymhvyj1b9g8b8mg0y-ibus-1.5.28-dev/include/ibus-1.0
@@ -0,0 +1 @@
+/nix/store/1s7z43yp594lrz7ymhvyj1b9g8b8mg0y-ibus-1.5.28-dev/include/ibus-1.0
\ No newline at end of file
diff --git a/a/external/ibus/nix/store/hvpil7yyqfiv87rgzvzjrq41daa182my-ibus-1.5.28-dev/include/ibus-1.0 b/a/external/ibus/nix/store/hvpil7yyqfiv87rgzvzjrq41daa182my-ibus-1.5.28-dev/include/ibus-1.0
deleted file mode 120000
index 54e99ab..0000000
--- a/a/external/ibus/nix/store/hvpil7yyqfiv87rgzvzjrq41daa182my-ibus-1.5.28-dev/include/ibus-1.0
+++ /dev/null
@@ -1 +0,0 @@
-/nix/store/hvpil7yyqfiv87rgzvzjrq41daa182my-ibus-1.5.28-dev/include/ibus-1.0
\ No newline at end of file
diff --git a/b/external/ibus/nix/store/kl4j6fjadg3lqaqv8r9p1lsjrwmi4cz8-glib-2.78.3/lib/glib-2.0/include b/b/external/ibus/nix/store/kl4j6fjadg3lqaqv8r9p1lsjrwmi4cz8-glib-2.78.3/lib/glib-2.0/include
new file mode 120000
index 0000000..d40e5f5
--- /dev/null
+++ b/b/external/ibus/nix/store/kl4j6fjadg3lqaqv8r9p1lsjrwmi4cz8-glib-2.78.3/lib/glib-2.0/include
@@ -0,0 +1 @@
+/nix/store/kl4j6fjadg3lqaqv8r9p1lsjrwmi4cz8-glib-2.78.3/lib/glib-2.0/include
\ No newline at end of file
diff --git a/a/external/ibus/nix/store/nynq8wryq2lzq92mcj0niiwady33qhss-glib-2.78.3/lib/glib-2.0/include b/a/external/ibus/nix/store/nynq8wryq2lzq92mcj0niiwady33qhss-glib-2.78.3/lib/glib-2.0/include
deleted file mode 120000
index 98c4e9f..0000000
--- a/a/external/ibus/nix/store/nynq8wryq2lzq92mcj0niiwady33qhss-glib-2.78.3/lib/glib-2.0/include
+++ /dev/null
@@ -1 +0,0 @@
-/nix/store/nynq8wryq2lzq92mcj0niiwady33qhss-glib-2.78.3/lib/glib-2.0/include
\ No newline at end of file
diff --git a/a/external/ibus/nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include b/a/external/ibus/nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include
deleted file mode 120000
index e3a34a7..0000000
--- a/a/external/ibus/nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include
+++ /dev/null
@@ -1 +0,0 @@
-/nix/store/vdaqdd9l32mnpxlv645y4sjslm82yv9x-glib-2.78.3-dev/include
\ No newline at end of file
diff --git a/a/external/qt_linux/BUILD.bazel b/b/external/qt_linux/BUILD.bazel
index 8c28eb8..2a29adf 100755
--- a/a/external/qt_linux/BUILD.bazel
+++ b/b/external/qt_linux/BUILD.bazel
@@ -6,11 +6,11 @@ package(
 cc_library(
     name = "qt_linux",
     hdrs = glob([
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/**",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/QtCore/**",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/QtGui/**",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/QtWidgets/**",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/mkspecs/linux-g++/**"
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/**",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/QtCore/**",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/QtGui/**",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/QtWidgets/**",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/mkspecs/linux-g++/**"
     ]),
     copts = [
         "-DQT_CORE_LIB",
@@ -18,11 +18,11 @@ cc_library(
         "-DQT_WIDGETS_LIB"
     ],
     includes = [
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/QtCore",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/QtGui",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include/QtWidgets",
-        "nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/mkspecs/linux-g++"
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/QtCore",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/QtGui",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include/QtWidgets",
+        "nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/mkspecs/linux-g++"
     ],
     linkopts = [
         "-lQt6Core",
diff --git a/a/external/qt_linux/libexec b/b/external/qt_linux/libexec
index 933c534..ae3cd8b 120000
--- a/a/external/qt_linux/libexec
+++ b/b/external/qt_linux/libexec
@@ -1 +1 @@
-/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/libexec
\ No newline at end of file
+/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/libexec
\ No newline at end of file
diff --git a/b/external/qt_linux/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include b/b/external/qt_linux/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include
new file mode 120000
index 0000000..5659261
--- /dev/null
+++ b/b/external/qt_linux/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include
@@ -0,0 +1 @@
+/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/include
\ No newline at end of file
diff --git a/b/external/qt_linux/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/mkspecs/linux-g++ b/b/external/qt_linux/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/mkspecs/linux-g++
new file mode 120000
index 0000000..655dba7
--- /dev/null
+++ b/b/external/qt_linux/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/mkspecs/linux-g++
@@ -0,0 +1 @@
+/nix/store/k5mav3nynvd2jzyv9rdvaky1ifpp0a5f-qtbase-6.6.1/mkspecs/linux-g++
\ No newline at end of file
diff --git a/a/external/qt_linux/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include b/a/external/qt_linux/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include
deleted file mode 120000
index 2a84ecc..0000000
--- a/a/external/qt_linux/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include
+++ /dev/null
@@ -1 +0,0 @@
-/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/include
\ No newline at end of file
diff --git a/a/external/qt_linux/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/mkspecs/linux-g++ b/a/external/qt_linux/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/mkspecs/linux-g++
deleted file mode 120000
index 764d447..0000000
--- a/a/external/qt_linux/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/mkspecs/linux-g++
+++ /dev/null
@@ -1 +0,0 @@
-/nix/store/r3d3wnmh480g9zy5pwj1s8qwljv3ppqk-qtbase-6.6.1/mkspecs/linux-g++
\ No newline at end of file
```
</details>

To me this seems like a bug in `buildBazelPackage`. I will see if there's any easy way I can fix this, but we may have to wait for the merge of this PR.

pineapplehunter commented:
I think I figured it out.
The problem was that bazel deps have some files referencing buildInputs, so when there's an update to any of the inputs (in this case: ibus, qt6.qtbase) it will try to access the old inputs and break on build. I assume this was the reason why aarch64 builds were failing too, because of different inputs for different architectures.
Fortunatly, those references could be generated on build so they could be deleted on fetchAttrs.preInstall.
I will update the PR with the patch once I can confirm it builds.

kirillrdy commented on pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix:
```diff
@@ -1,89 +1,73 @@
-    export GYP_DEFINES="document_dir=$out/share/doc/mozc use_libzinnia=1 use_libprotobuf=1 ibus_mozc_path=$out/lib/ibus-mozc/ibus-engine-mozc zinnia_model_file=${tegaki-zinnia-japanese}/share/tegaki/models/zinnia/handwriting-ja.model"
-    cd src && python build_mozc.py gyp --gypdir=${python3Packages.gyp}/bin --server_dir=$out/lib/mozc
+  fetchAttrs = {
+    sha256 = "sha256-17QHh1MJUu8OK/T+WSpLXEx83DmRORLN7yLzILqP7vw=";
```
I recommend to add     `disallowedReferences = [ "/nix/store" ]; `

this check fails if you remove preInstall

kirillrdy replied:
on a second though, maybe its better to add it to buildBazelPackage depending how much more work we want to do. I'll leave it upto you to decide

jtojnar replied:
I think empty [`allowedReferences`](https://nixos.org/manual/nix/stable/language/advanced-attributes.html#adv-attr-allowedReferences) is more idiomatic. I have not seen disallowing whole `/nix/store` documented anywhere.

I do not think it is feasible to do it in the outer derivation, since there are ELF files that link other libraries.

kirillrdy replied:
we are trying to do it on FOD, to prevent references to build inputs. what would empty allowedReferences do ?

kirillrdy replied:
appologies, i just read documentation :-) cc @pineapplehunter

pineapplehunter replied:
Because the result is in a tar.gz it seems like it's not referencing anything.

```shell
# this only shows one reference
$ nix path-info ".#ibus-engines.mozc.deps" -r
/nix/store/p75bdk45s2ncjfqskhh8xwp47103hz49-ibus-mozc-2.29.5268.102-deps.tar.gz
```

Adding allowedReferences to fetchAttrs didn't seem to help in this case. Is there any simple way to check for references in maybe postInstal?

ncfavier replied:
```
error: hash mismatch in fixed-output derivation '/nix/store/9l16176d0m63j51x6ga10mlid605dzkq-ibus-mozc-2.29.5268.102-deps.tar.gz.drv':
         specified: sha256-17QHh1MJUu8OK/T+WSpLXEx83DmRORLN7yLzILqP7vw=
            got:    sha256-cONjTaGQcCMhfjLCPa2BSLE9/bVHvFOuv06Pg+wlJm4=
```

pineapplehunter replied:
Thanks!
So this time, it seems that some of the bazel deps (`zip_code_ken_all` and `zip_code_jigyosyo`) changed. It was in the logs before, but I assumed those files don't change so frequently.
I will try to do something like what's done in #251706.

pineapplehunter replied:
This problem should be fixed now. I replaced all the http urls with a file path from a github repo.
Let's wait a few more days and check again if anything fails.

kirillrdy reviewed (approved):
@pineapplehunter i am happy to merge as is, but lets leave it for a day or two maybe someone can suggest something

kirillrdy commented on pkgs/tools/inputmethods/ibus-engines/ibus-mozc/default.nix:
```diff
@@ -1,89 +1,86 @@
+  postPatch = ''
+    substituteInPlace src/config.bzl \
+      --replace-fail "/usr/bin/xdg-open" "${xdg-utils}/bin/xdg-open" \
+      --replace-fail "/usr" "$out"
```
:+1: for `--replace-fail`

kirillrdy reviewed (commented):
Result of `nixpkgs-review pr 282148` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>7 packages built:</summary>
  <ul>
    <li>emacsPackages.ac-mozc</li>
    <li>emacsPackages.mozc</li>
    <li>emacsPackages.mozc-cand-posframe</li>
    <li>emacsPackages.mozc-im</li>
    <li>emacsPackages.mozc-popup</li>
    <li>emacsPackages.mozc-temp</li>
    <li>ibus-engines.mozc</li>
  </ul>
</details>

kirillrdy commented:
@pineapplehunter thanks again !

TomaSajt commented:
Hello from the future!

This PR dropped `mozc`'s dependency on `zinnia` and `tegaki-zinnia-japanese`.
AFAICT both packages [were added to nixpkgs](https://github.com/NixOS/nixpkgs/pull/7104) by @gebner alongside `mozc` because it depended on it at the time.

Both of those packages have become leaf packages after this PR.

Is anyone against dropping them?

None of the two packages seem to have any other usages in public github repos, according to github search.

---
Note: I discovered these packages after seeing that `zinnia`'s build was [failing on clang](https://hydra.nixos.org/build/296515810).
There is a pretty easy fix, though, so we could just apply that, instead of dropping the package.

kirillrdy commented:
I don't have strong option for these specific packages, but looking at https://repology.org/projects/?search=zinnia looks like it packaged in many places, so if not too hard to fix, I'd keep them

TomaSajt commented:
Okay, thanks for your response. I'll just open a fix PR, then.
